infocmdb/attribute_group.go

```diff
@@ -0,0 +1,145 @@
+package infocmdb
+
+import (
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	v2 "github.com/infonova/infocmdb-sdk-go/infocmdb/v2/infocmdb"
+	utilError "github.com/infonova/infocmdb-sdk-go/util/error"
+)
+
+var boolToString = map[bool]string{false: "0", true: "1"}
+
+type getAttributeGroupIdValue struct {
+	Data []struct {
+		GroupId int `json:"id,string"`
+	} `json:"data"`
+}
+
+func (c *Client) GetAttributeGroupIdByName(attributeGroupName string) (attGroupId int, err error) {
```

Comment:
I don't know how often the function is called in your workflow, but I wouldn't be surprised if its called often to create multiple attributes all belonging to the same attribute group.
We might want to cache the return value to improve the workflow performance and reduce the load on the cmdb in the future.

Reply:
Added cacheing:

cacheKey := "GetAttributeGroupIdByName" + attributeGroupName
	cached, found := c.v1.Cache.Get(cacheKey)
	if found {
		return cached.(int), nil
	}